HousingSearchApi.Tests/V1/TestHelpers/CreateRandomISOStartOfTenureDate.cs

```diff
@@ -0,0 +1,32 @@
+using AutoFixture;
+using System;
+using QueryableTenure = Hackney.Shared.HousingSearch.Gateways.Models.Tenures.QueryableTenure;
+
+namespace HousingSearchApi.Tests.V1.TestHelpers
+{
+    public class CreateRandomISOStartOfTenureDate : ICustomization
+    {
+        public void Customize(IFixture fixture)
+        {
+            fixture.Customize<QueryableTenure>(x =>
+                x.With(t => t.StartOfTenureDate, GenerateRandomTenureStartDate()));
+        }
+
+        private string GenerateRandomTenureStartDate()
+        {
+            //pick a sensible random day in the past (today inclusive)
+            var addDays = new Random().Next(TenureStartDateAddDaysMinValue, TenureStartDateAddDaysMaxValue);
+
+            return DateTime.UtcNow.AddDays(addDays).ToString("yyyy-MM-ddTHH:mm:ssZ");
+        }
+        public static int TenureStartDateAddDaysMinValue
+        {
+            get { return -3000; }
+        }
+
+        public static int TenureStartDateAddDaysMaxValue
+        {
+            get { return 1; }
+        }
+    }
+}
```

Comment:
These fixture add a lot of complexity to the project.

Is there any reason, why the `Bogus` package could not be used?
This package is already installed under this tests project: https://github.com/LBHackney-IT/housing-search-api/blob/93775770a15dd5a397f8872aa270c09c6d9d2aaf/HousingSearchApi.Tests/HousingSearchApi.Tests.csproj#L55

And has all the fancy random date customisation logic you could ask:

![image](https://github.com/LBHackney-IT/housing-search-api/assets/43747286/24e4eb35-d81b-4dc5-a0cf-b31278bebc15)

Reply:
You use much the same as Autofixture:
```cs
// at the top of the file:
using Bogus;

// at the top of the class:
Faker _faker = new Faker();

// within your test arrange step:
var tenureStartDate = _faker.Date.Soon();
```

Reply:
In your case it would be:
``` cs
var tenure = _fixture()
    .Build<QueryableTenure>()
    .With(x => x.StartOfTenureDate,  _faker.Date.Soon())
    .Create() ;
```

Reply:
Yes, I guess that's a bit overly complex and reinvent some existing wheels. Because we want those dates to be in exact string format and also in the past, I can effectively use `_faker.Date.Past().ToString("yyyy-MM-ddTHH:mm:ssZ")` whenever we need to create a new fixture. That should be sufficient for the purpose of these test. I can refactor this to get rid of the customization altogether. And yes I'll look at the fixture setup, so there are no magic string all over.

Reply:
New fixture setup to handle the date implemented in [642e7e2](https://github.com/LBHackney-IT/housing-search-api/pull/224/commits/642e7e23a648dbb325df8c2dc925aca7c33fffae)

Reply:
Good one!